enhancements/authentication/invalid-certs.md

```diff
@@ -0,0 +1,152 @@
+---
+title: invalid-certs
+authors:
+  - "@s-urbaniak"
+reviewers:
+  - "@sttts"
+  - "@stlaz"
+  - "@deads2k"
+approvers:
+  - "@mfojtik"
+creation-date: 2022-12-08
+last-updated: 2022-12-08
+---
+
+# Detecting invalid CA certificates
+
+## Summary
+
+OpenShift 4.10 is going to be rebased against Kubernetes 1.23. This requires using Go 1.17.
+However, starting with Go 1.17 support for legacy invalid is going to be removed, see https://go.dev/doc/go1.17.
+More formally, the temporary `GODEBUG=x509ignoreCN=0` flag has been removed.
+This implies that starting from OpenShift 4.10 invalid certificates will not be trusted any more
+as they will fail verification.
+
+This enhancement suggests a solution to prevent broken workloads from OpenShift 4.9 to OpenShift 4.10
+if invalid certificates are detected.
+
+This enhancment therefore proposes:
+- a way to detect invalid certificates in kube-apiserver and oauth-server
+- a way to prevent upgrades from OpenShift 4.9 to OpenShift 4.10 in the face of invalid certificates
+
+## Motivation
+
+We want to prevent broken workloads due to invalid certificates
+when upgrading from OpenShift 4.9 to OpenShift 4.10.
+
+### Goals
+
+A means to detect invalid certificates for TLS connections between:
+- kube-apiserver and API webhooks
+- kube-apiserver and aggregated API endpoints
+- oauth-server and its external route
+- oauth-server and external identity providers
+
```

Comment:
custom serving certs for kube-apiserver

Reply:
@sttts do you have a place where I can hook into the verification of those? Currently, I think we'd need a separate controller in `cluster-kube-apiserver-operator` that checks `config.openshift.io.APIServer#servingInfo.namedCertificates`, parses them, and verifies for invalid certs.

Reply:
I believe @sanchezl worked on that. He can link the code here.